File: Data Structures & Algorithms/sliding-window-maximum/submission-2.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        maxList = []

        for l in range(len(nums) - k + 1):
            maxElement = float('-inf')
            r = l + k
            for i in range(l, r):
                maxElement = max(maxElement, nums[i])
            maxList.append(maxElement)
        return maxList

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> sqn;
        int res = 0;

        for (int i : nums) {
            if (!sqn[i]) {
                sqn[i] = sqn[i - 1] + sqn[i + 1] + 1;
                sqn[i - sqn[i - 1]] = sqn[i];
                sqn[i + sqn[i + 1]] = sqn[i];
                res = max(res, sqn[i]);
            }
        }
        return res;
    }
};
